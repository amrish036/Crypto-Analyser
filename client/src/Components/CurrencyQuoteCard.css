.Card{
    display: flex;
    flex-Direction:column;
    margin: 50px 
}

.CardHeading{
    color:red;
}

