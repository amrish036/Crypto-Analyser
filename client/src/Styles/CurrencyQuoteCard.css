.Card{
    display: flex;
    flex-Direction:column;
    margin: 20px 10px ;
}

.CardHeading{
    color:red;
}

