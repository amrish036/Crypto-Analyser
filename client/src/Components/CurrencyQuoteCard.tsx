import React, { Component } from 'react';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import Quote from '../Models/Quote';
import Currency from '../Models/Currency';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import moment from 'moment';
import './CurrencyQuoteCard.css'

class CurrencyQuoteCard extends React.Component {


    constructor(props: any) {
        super(props);
    }
    componentDidMount() {
        const Quote = this.props;
        const BestQuote = '';
        console.log("Component Mounted")
    }

    render() {
        const Quotes: any = this.props;
        const firstQuote = Quotes.quotes[0].quotes[0];
        const bestQuote = GetBestQuote(Quotes.quotes);
        const profit = (bestQuote.price - firstQuote.price).toFixed(2);
        // const BTC: any = Quotes.quotes.filter((x: Currency) => x.currency === 'BTC');
        // const ETC: any = Quotes.quotes.filter((x: Currency) => x.currency === 'ETC');
        // const LTC: any = Quotes.quotes.filter((x: Currency) => x.currency === 'LTC');

        return (
            <div className="Card">
                <Card>
                    <CardContent>
                        <Typography className="CardHeading" gutterBottom variant="h5" component="h2" color="primary">
                            {convertToDay(Quotes.quotes[0].date)} <br />
                            {Quotes.quotes[0].currency} <br />
                        </Typography>
                        <Table >
                            <TableHead>
                                <TableRow>
                                    <TableCell>BUY</TableCell>
                                    <TableCell>SELL</TableCell>
                                </TableRow>
                            </TableHead>
                            <TableBody>
                                <TableRow>
                                    <TableCell>${firstQuote.price}</TableCell>
                                    <TableCell>${bestQuote.price}</TableCell>
                                </TableRow>
                                <TableRow>
                                    <TableCell>{ConvertToTime(firstQuote.time)}</TableCell>
                                    <TableCell>{ConvertToTime(bestQuote.time)}</TableCell>
                                </TableRow>
                            </TableBody>
                        </Table>
                        <br/>
                        <Typography gutterBottom variant="h5" component="h3" style={{color:'teal'}} >
                        {ShowProfit(parseFloat(profit))}
                        </Typography>
                    </CardContent>

                    {/* {console.log(Quotes.quotes)}
          {console.log(BTC[0].currency)}
          {console.log(LTC[0].currency)}
          {console.log(ETC[0].date)}
          {console.log(convertToDay(ETC[0].date))}
          {console.log(ETC[0].currency)} <br/> */}
                </Card>
            </div>
        );
    }
}

function convertToDay(dateString: string) {
    var convertedDate = moment(dateString);
    console.log(convertedDate);
    return convertedDate.format('DD-MMM-YYYY');
}

function ConvertToTime(timeString: string) {
    var convertedTime = moment(timeString, "hhmm").format("h:mm A");
    console.log(convertedTime);
    return convertedTime;
}


function GetBestQuote(allQuotes: any) {
    var allQuotes = allQuotes[0].quotes.slice(1);
    var numberOfQuotes = allQuotes.length;
    var sortedQuotes = allQuotes.map((x: any) => parseFloat(x.price)).sort();
    var highestQuote = sortedQuotes[numberOfQuotes - 1];
    var bestQuote = allQuotes.filter((x:any)=>parseFloat(x.price) === highestQuote);
    return bestQuote[0];
}

function ShowProfit(profit: number){
if(profit<0){
    return<span style={{color:'red'}}>$Profit: {profit.toLocaleString()}</span>
}
else{
    return <span>$Profit: {profit.toLocaleString()}</span>
}
}
function GetProfit(allQuotes: any) {
    {
        var allQuotes = allQuotes[0].quotes.slice(1);
        var numberOfQuotes = allQuotes.length;
        var sortedQuotes = allQuotes.map((x: any) => parseFloat(x.price)).sort();
        var highestQuote = sortedQuotes[numberOfQuotes - 1];
        var bestQuote = allQuotes.filter((x: any) => x.price === highestQuote.toString());
        return bestQuote[0];
    }
}

export default CurrencyQuoteCard;