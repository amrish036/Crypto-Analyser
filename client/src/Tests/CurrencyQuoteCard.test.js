import React from 'react';
import ReactDOM from 'react-dom';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import CurrencyQuoteCard from '../Components/CurrencyQuoteCard';

configure({ adapter: new Adapter() });

describe('<CurrencyCardQuote />', () => {
    let wrapper = beforeEach(() => { wrapper = shallow(<CurrencyQuoteCard quotes={HistoricalData} />) });

    it('includes one card div', () => {
        console.log(wrapper.find('div.Card').text());
        expect(wrapper.find('div.Card')).toHaveLength(1)
    });
})


it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CurrencyQuoteCard quotes={HistoricalData} />, div)
    ReactDOM.unmountComponentAtNode(div);
})

it('should match the currencyQuoteCard snapshot', () => {
    const component = shallow(<CurrencyQuoteCard quotes={HistoricalData} />);
    expect(component).toMatchSnapshot();
})

const HistoricalData = {
    "currency": "BTC",
    "date": "20181107",
    "quotes": [
        {
            "time": "0915",
            "price": "34.98"
        },
        {
            "time": "1045",
            "price": "36.13"
        },
        {
            "time": "1230",
            "price": "37.01"
        },
        {
            "time": "1400",
            "price": "95.98"
        },
        {
            "time": "1530",
            "price": "39.56"
        }
    ]
}