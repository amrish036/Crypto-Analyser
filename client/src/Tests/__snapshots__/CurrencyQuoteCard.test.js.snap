// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the currencyQuoteCard snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrencyQuoteCard
    quotes={
      Object {
        "currency": "BTC",
        "date": "20181107",
        "quotes": Array [
          Object {
            "price": "34.98",
            "time": "0915",
          },
          Object {
            "price": "36.13",
            "time": "1045",
          },
          Object {
            "price": "37.01",
            "time": "1230",
          },
          Object {
            "price": "95.98",
            "time": "1400",
          },
          Object {
            "price": "39.56",
            "time": "1530",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Card)>
        <WithStyles(CardContent)>
          <WithStyles(Typography)
            className="CardHeading"
            color="primary"
            component="h2"
            gutterBottom={true}
            variant="h5"
          >
            07-Nov-2018
             
            <br />
            BTC
             
            <br />
          </WithStyles(Typography)>
          <WithStyles(Table)>
            <WithStyles(TableHead)>
              <WithStyles(TableRow)>
                <WithStyles(TableCell)>
                  BUY
                </WithStyles(TableCell)>
                <WithStyles(TableCell)>
                  SELL
                </WithStyles(TableCell)>
              </WithStyles(TableRow)>
            </WithStyles(TableHead)>
            <WithStyles(TableBody)>
              <WithStyles(TableRow)>
                <WithStyles(TableCell)>
                  $
                  34.98
                </WithStyles(TableCell)>
                <WithStyles(TableCell)>
                  $
                  95.98
                </WithStyles(TableCell)>
              </WithStyles(TableRow)>
              <WithStyles(TableRow)>
                <WithStyles(TableCell)>
                  9:15 AM
                </WithStyles(TableCell)>
                <WithStyles(TableCell)>
                  2:00 PM
                </WithStyles(TableCell)>
              </WithStyles(TableRow)>
            </WithStyles(TableBody)>
          </WithStyles(Table)>
          <br />
          <WithStyles(Typography)
            component="h3"
            gutterBottom={true}
            style={
              Object {
                "color": "teal",
              }
            }
            variant="h5"
          >
            <span>
              Profit: $
              61
            </span>
          </WithStyles(Typography)>
        </WithStyles(CardContent)>
      </WithStyles(Card)>,
      "className": "Card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(CardContent)>
          <WithStyles(Typography)
            className="CardHeading"
            color="primary"
            component="h2"
            gutterBottom={true}
            variant="h5"
          >
            07-Nov-2018
             
            <br />
            BTC
             
            <br />
          </WithStyles(Typography)>
          <WithStyles(Table)>
            <WithStyles(TableHead)>
              <WithStyles(TableRow)>
                <WithStyles(TableCell)>
                  BUY
                </WithStyles(TableCell)>
                <WithStyles(TableCell)>
                  SELL
                </WithStyles(TableCell)>
              </WithStyles(TableRow)>
            </WithStyles(TableHead)>
            <WithStyles(TableBody)>
              <WithStyles(TableRow)>
                <WithStyles(TableCell)>
                  $
                  34.98
                </WithStyles(TableCell)>
                <WithStyles(TableCell)>
                  $
                  95.98
                </WithStyles(TableCell)>
              </WithStyles(TableRow)>
              <WithStyles(TableRow)>
                <WithStyles(TableCell)>
                  9:15 AM
                </WithStyles(TableCell)>
                <WithStyles(TableCell)>
                  2:00 PM
                </WithStyles(TableCell)>
              </WithStyles(TableRow)>
            </WithStyles(TableBody)>
          </WithStyles(Table)>
          <br />
          <WithStyles(Typography)
            component="h3"
            gutterBottom={true}
            style={
              Object {
                "color": "teal",
              }
            }
            variant="h5"
          >
            <span>
              Profit: $
              61
            </span>
          </WithStyles(Typography)>
        </WithStyles(CardContent)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              className="CardHeading"
              color="primary"
              component="h2"
              gutterBottom={true}
              variant="h5"
            >
              07-Nov-2018
               
              <br />
              BTC
               
              <br />
            </WithStyles(Typography)>,
            <WithStyles(Table)>
              <WithStyles(TableHead)>
                <WithStyles(TableRow)>
                  <WithStyles(TableCell)>
                    BUY
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                    SELL
                  </WithStyles(TableCell)>
                </WithStyles(TableRow)>
              </WithStyles(TableHead)>
              <WithStyles(TableBody)>
                <WithStyles(TableRow)>
                  <WithStyles(TableCell)>
                    $
                    34.98
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                    $
                    95.98
                  </WithStyles(TableCell)>
                </WithStyles(TableRow)>
                <WithStyles(TableRow)>
                  <WithStyles(TableCell)>
                    9:15 AM
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                    2:00 PM
                  </WithStyles(TableCell)>
                </WithStyles(TableRow)>
              </WithStyles(TableBody)>
            </WithStyles(Table)>,
            <br />,
            <WithStyles(Typography)
              component="h3"
              gutterBottom={true}
              style={
                Object {
                  "color": "teal",
                }
              }
              variant="h5"
            >
              <span>
                Profit: $
                61
              </span>
            </WithStyles(Typography)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "07-Nov-2018",
                " ",
                <br />,
                "BTC",
                " ",
                <br />,
              ],
              "className": "CardHeading",
              "color": "primary",
              "component": "h2",
              "gutterBottom": true,
              "variant": "h5",
            },
            "ref": null,
            "rendered": Array [
              "07-Nov-2018",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "BTC",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(TableHead)>
                  <WithStyles(TableRow)>
                    <WithStyles(TableCell)>
                      BUY
                    </WithStyles(TableCell)>
                    <WithStyles(TableCell)>
                      SELL
                    </WithStyles(TableCell)>
                  </WithStyles(TableRow)>
                </WithStyles(TableHead)>,
                <WithStyles(TableBody)>
                  <WithStyles(TableRow)>
                    <WithStyles(TableCell)>
                      $
                      34.98
                    </WithStyles(TableCell)>
                    <WithStyles(TableCell)>
                      $
                      95.98
                    </WithStyles(TableCell)>
                  </WithStyles(TableRow)>
                  <WithStyles(TableRow)>
                    <WithStyles(TableCell)>
                      9:15 AM
                    </WithStyles(TableCell)>
                    <WithStyles(TableCell)>
                      2:00 PM
                    </WithStyles(TableCell)>
                  </WithStyles(TableRow)>
                </WithStyles(TableBody)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(TableRow)>
                    <WithStyles(TableCell)>
                      BUY
                    </WithStyles(TableCell)>
                    <WithStyles(TableCell)>
                      SELL
                    </WithStyles(TableCell)>
                  </WithStyles(TableRow)>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <WithStyles(TableCell)>
                        BUY
                      </WithStyles(TableCell)>,
                      <WithStyles(TableCell)>
                        SELL
                      </WithStyles(TableCell)>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "BUY",
                      },
                      "ref": null,
                      "rendered": "BUY",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "SELL",
                      },
                      "ref": null,
                      "rendered": "SELL",
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <WithStyles(TableRow)>
                      <WithStyles(TableCell)>
                        $
                        34.98
                      </WithStyles(TableCell)>
                      <WithStyles(TableCell)>
                        $
                        95.98
                      </WithStyles(TableCell)>
                    </WithStyles(TableRow)>,
                    <WithStyles(TableRow)>
                      <WithStyles(TableCell)>
                        9:15 AM
                      </WithStyles(TableCell)>
                      <WithStyles(TableCell)>
                        2:00 PM
                      </WithStyles(TableCell)>
                    </WithStyles(TableRow)>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        <WithStyles(TableCell)>
                          $
                          34.98
                        </WithStyles(TableCell)>,
                        <WithStyles(TableCell)>
                          $
                          95.98
                        </WithStyles(TableCell)>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": Array [
                            "$",
                            "34.98",
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          "$",
                          "34.98",
                        ],
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": Array [
                            "$",
                            "95.98",
                          ],
                        },
                        "ref": null,
                        "rendered": Array [
                          "$",
                          "95.98",
                        ],
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        <WithStyles(TableCell)>
                          9:15 AM
                        </WithStyles(TableCell)>,
                        <WithStyles(TableCell)>
                          2:00 PM
                        </WithStyles(TableCell)>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "9:15 AM",
                        },
                        "ref": null,
                        "rendered": "9:15 AM",
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "2:00 PM",
                        },
                        "ref": null,
                        "rendered": "2:00 PM",
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span>
                Profit: $
                61
              </span>,
              "component": "h3",
              "gutterBottom": true,
              "style": Object {
                "color": "teal",
              },
              "variant": "h5",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Profit: $",
                  "61",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Profit: $",
                "61",
              ],
              "type": "span",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Card)>
          <WithStyles(CardContent)>
            <WithStyles(Typography)
              className="CardHeading"
              color="primary"
              component="h2"
              gutterBottom={true}
              variant="h5"
            >
              07-Nov-2018
               
              <br />
              BTC
               
              <br />
            </WithStyles(Typography)>
            <WithStyles(Table)>
              <WithStyles(TableHead)>
                <WithStyles(TableRow)>
                  <WithStyles(TableCell)>
                    BUY
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                    SELL
                  </WithStyles(TableCell)>
                </WithStyles(TableRow)>
              </WithStyles(TableHead)>
              <WithStyles(TableBody)>
                <WithStyles(TableRow)>
                  <WithStyles(TableCell)>
                    $
                    34.98
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                    $
                    95.98
                  </WithStyles(TableCell)>
                </WithStyles(TableRow)>
                <WithStyles(TableRow)>
                  <WithStyles(TableCell)>
                    9:15 AM
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                    2:00 PM
                  </WithStyles(TableCell)>
                </WithStyles(TableRow)>
              </WithStyles(TableBody)>
            </WithStyles(Table)>
            <br />
            <WithStyles(Typography)
              component="h3"
              gutterBottom={true}
              style={
                Object {
                  "color": "teal",
                }
              }
              variant="h5"
            >
              <span>
                Profit: $
                61
              </span>
            </WithStyles(Typography)>
          </WithStyles(CardContent)>
        </WithStyles(Card)>,
        "className": "Card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(CardContent)>
            <WithStyles(Typography)
              className="CardHeading"
              color="primary"
              component="h2"
              gutterBottom={true}
              variant="h5"
            >
              07-Nov-2018
               
              <br />
              BTC
               
              <br />
            </WithStyles(Typography)>
            <WithStyles(Table)>
              <WithStyles(TableHead)>
                <WithStyles(TableRow)>
                  <WithStyles(TableCell)>
                    BUY
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                    SELL
                  </WithStyles(TableCell)>
                </WithStyles(TableRow)>
              </WithStyles(TableHead)>
              <WithStyles(TableBody)>
                <WithStyles(TableRow)>
                  <WithStyles(TableCell)>
                    $
                    34.98
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                    $
                    95.98
                  </WithStyles(TableCell)>
                </WithStyles(TableRow)>
                <WithStyles(TableRow)>
                  <WithStyles(TableCell)>
                    9:15 AM
                  </WithStyles(TableCell)>
                  <WithStyles(TableCell)>
                    2:00 PM
                  </WithStyles(TableCell)>
                </WithStyles(TableRow)>
              </WithStyles(TableBody)>
            </WithStyles(Table)>
            <br />
            <WithStyles(Typography)
              component="h3"
              gutterBottom={true}
              style={
                Object {
                  "color": "teal",
                }
              }
              variant="h5"
            >
              <span>
                Profit: $
                61
              </span>
            </WithStyles(Typography)>
          </WithStyles(CardContent)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                className="CardHeading"
                color="primary"
                component="h2"
                gutterBottom={true}
                variant="h5"
              >
                07-Nov-2018
                 
                <br />
                BTC
                 
                <br />
              </WithStyles(Typography)>,
              <WithStyles(Table)>
                <WithStyles(TableHead)>
                  <WithStyles(TableRow)>
                    <WithStyles(TableCell)>
                      BUY
                    </WithStyles(TableCell)>
                    <WithStyles(TableCell)>
                      SELL
                    </WithStyles(TableCell)>
                  </WithStyles(TableRow)>
                </WithStyles(TableHead)>
                <WithStyles(TableBody)>
                  <WithStyles(TableRow)>
                    <WithStyles(TableCell)>
                      $
                      34.98
                    </WithStyles(TableCell)>
                    <WithStyles(TableCell)>
                      $
                      95.98
                    </WithStyles(TableCell)>
                  </WithStyles(TableRow)>
                  <WithStyles(TableRow)>
                    <WithStyles(TableCell)>
                      9:15 AM
                    </WithStyles(TableCell)>
                    <WithStyles(TableCell)>
                      2:00 PM
                    </WithStyles(TableCell)>
                  </WithStyles(TableRow)>
                </WithStyles(TableBody)>
              </WithStyles(Table)>,
              <br />,
              <WithStyles(Typography)
                component="h3"
                gutterBottom={true}
                style={
                  Object {
                    "color": "teal",
                  }
                }
                variant="h5"
              >
                <span>
                  Profit: $
                  61
                </span>
              </WithStyles(Typography)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "07-Nov-2018",
                  " ",
                  <br />,
                  "BTC",
                  " ",
                  <br />,
                ],
                "className": "CardHeading",
                "color": "primary",
                "component": "h2",
                "gutterBottom": true,
                "variant": "h5",
              },
              "ref": null,
              "rendered": Array [
                "07-Nov-2018",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                "BTC",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(TableHead)>
                    <WithStyles(TableRow)>
                      <WithStyles(TableCell)>
                        BUY
                      </WithStyles(TableCell)>
                      <WithStyles(TableCell)>
                        SELL
                      </WithStyles(TableCell)>
                    </WithStyles(TableRow)>
                  </WithStyles(TableHead)>,
                  <WithStyles(TableBody)>
                    <WithStyles(TableRow)>
                      <WithStyles(TableCell)>
                        $
                        34.98
                      </WithStyles(TableCell)>
                      <WithStyles(TableCell)>
                        $
                        95.98
                      </WithStyles(TableCell)>
                    </WithStyles(TableRow)>
                    <WithStyles(TableRow)>
                      <WithStyles(TableCell)>
                        9:15 AM
                      </WithStyles(TableCell)>
                      <WithStyles(TableCell)>
                        2:00 PM
                      </WithStyles(TableCell)>
                    </WithStyles(TableRow)>
                  </WithStyles(TableBody)>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(TableRow)>
                      <WithStyles(TableCell)>
                        BUY
                      </WithStyles(TableCell)>
                      <WithStyles(TableCell)>
                        SELL
                      </WithStyles(TableCell)>
                    </WithStyles(TableRow)>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        <WithStyles(TableCell)>
                          BUY
                        </WithStyles(TableCell)>,
                        <WithStyles(TableCell)>
                          SELL
                        </WithStyles(TableCell)>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "BUY",
                        },
                        "ref": null,
                        "rendered": "BUY",
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "SELL",
                        },
                        "ref": null,
                        "rendered": "SELL",
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <WithStyles(TableRow)>
                        <WithStyles(TableCell)>
                          $
                          34.98
                        </WithStyles(TableCell)>
                        <WithStyles(TableCell)>
                          $
                          95.98
                        </WithStyles(TableCell)>
                      </WithStyles(TableRow)>,
                      <WithStyles(TableRow)>
                        <WithStyles(TableCell)>
                          9:15 AM
                        </WithStyles(TableCell)>
                        <WithStyles(TableCell)>
                          2:00 PM
                        </WithStyles(TableCell)>
                      </WithStyles(TableRow)>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": Array [
                          <WithStyles(TableCell)>
                            $
                            34.98
                          </WithStyles(TableCell)>,
                          <WithStyles(TableCell)>
                            $
                            95.98
                          </WithStyles(TableCell)>,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "children": Array [
                              "$",
                              "34.98",
                            ],
                          },
                          "ref": null,
                          "rendered": Array [
                            "$",
                            "34.98",
                          ],
                          "type": [Function],
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "children": Array [
                              "$",
                              "95.98",
                            ],
                          },
                          "ref": null,
                          "rendered": Array [
                            "$",
                            "95.98",
                          ],
                          "type": [Function],
                        },
                      ],
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": Array [
                          <WithStyles(TableCell)>
                            9:15 AM
                          </WithStyles(TableCell)>,
                          <WithStyles(TableCell)>
                            2:00 PM
                          </WithStyles(TableCell)>,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "children": "9:15 AM",
                          },
                          "ref": null,
                          "rendered": "9:15 AM",
                          "type": [Function],
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "children": "2:00 PM",
                          },
                          "ref": null,
                          "rendered": "2:00 PM",
                          "type": [Function],
                        },
                      ],
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span>
                  Profit: $
                  61
                </span>,
                "component": "h3",
                "gutterBottom": true,
                "style": Object {
                  "color": "teal",
                },
                "variant": "h5",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Profit: $",
                    "61",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Profit: $",
                  "61",
                ],
                "type": "span",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
